02-nestjs-basics/01-tasks-controller/tasks/tasks.service.ts

```diff
@@ -4,14 +4,57 @@ import { Task } from "./task.model";
 @Injectable()
 export class TasksService {
   private tasks: Task[] = [];
+  private id: number = 1;
 
-  getAllTasks(): Task[] {}
+  getAllTasks(): Task[] {
+    return this.tasks;
+  }
 
-  getTaskById(id: string): Task {}
+  getTaskById(id: string): Task {
+    if (this.tasks.findIndex(task => task.id === id) !== -1) {
```

Comment:
наверное можно немного сократить запись
```suggestion
    const task = this.tasks.find((task) => task.id === id);
    if (!task) throw new NotFoundException(`Task with id ${id} not found`);
    return task;
```